hooks/chat/useBots.tsx

```
@@ -37,41 +37,42 @@ export const useBots = (roomName: string | undefined) => {
-          setCurrentBots((prev) => prev + 1)
-          console.debug(`Bot ${clientId} joined room ${roomName}`)
-
-          // Check if bot should publish messages
-          if (Math.random() > BOT_PUBLISHER_PROBABILITY) return
-
-          // Function to send message and reset interval for more human-like behavior
-          const sendMessageWithRandomInterval = () => {
-            room.messages
-              .send({ text: generateMessage() })
-              .then(() => {
-                console.log(`Bot ${clientId} sent message in room ${roomName}`)
-              })
-              .catch((error) => {
-                console.error(
-                  `Error sending message in room ${roomName}`,
-                  error
-                )
-              })
-            const randomInterval =
-              Math.floor(Math.random() * (BOT_INTERVAL - 10000)) + 10000 // Ensure minimum interval of 10 seconds
-            setTimeout(sendMessageWithRandomInterval, randomInterval)
-          }
-          sendMessageWithRandomInterval()
-        })
-        .catch((error) => {
-          setCurrentBots((prev) => prev + 1)
-          console.error(`Error joining room ${roomName}`, error)
-        })
+            // Function to send message and reset interval for more human-like behavior
+            const sendMessageWithRandomInterval = () => {
+              room.messages
+                .send({ text: generateMessage() })
+                .then(() => {
+                  console.log(
+                    `Bot ${clientId} sent message in room ${roomName}`
+                  )
+                })
+                .catch((error) => {
+                  console.error(
+                    `Error sending message in room ${roomName}`,
+                    error
+                  )
+                })
+              const randomInterval =
+                Math.floor(Math.random() * (BOT_INTERVAL - 10000)) + 10000 // Ensure minimum interval of 10 seconds
```

Min interval of 10s is true here only if BOT_INTERVAL is >10s. Must ensure `BOT_INTERVAL-10000` is positive. (I know this isn't a new thing)

---

Thats very true, though tbh it should happen at the parsing level when we read the env vars in, probably it should throw an error and tell you to set a correct value. Same would be true for BOT_COUNT, but as it's just a demo app and these are env vars we control, do you think its worth adding in?

---

I've added parsing checks now anyway, it was only a small change :)